source/_posts/2016-07-09-exporting-processing-data.markdown

```diff
@@ -0,0 +1,116 @@
+  Visualization with DB Browser for SQLite
+</p>
+
+You may ask: Why not do this with LibreOffice Calc or another spreadsheet application? As most spreadsheet applications are not able to work directly with SQLite database we are going to export the data from the database to [CSV](https://en.wikipedia.org/wiki/Comma-separated_values).
+
+```bash
+$ sqlite3 -header -csv /home/ha/.homeassistant/home-assistant_v2.db "SELECT last_changed, state FROM states WHERE entity_id = 'sensor.aare' AND last_changed BETWEEN '2016-07-05 00:00:00.000000' AND '2016-07-07 00:00:00.000000';" > sensor.csv
+```
+
+The ordering for the `SELECT` was changed to get the time stamps first and then the state. Now we can import the CSV file into the application of your choice, here it's LibreOffice Calc.
+
+<p class='img'>
+  <img src='/images/blog/2016-07-reporting/libreoffice-import.png' />
+  Import of the CSV file 
+</p>
+
+After the import a graph can be created over the existing data.
+
+<p class='img'>
+  <img src='/images/blog/2016-07-reporting/libreoffice-graph.png' />
+  Graph in LibreOffice
+</p>
+
+For the generation of graphs without spreadsheet application is [matplotlib](http://matplotlib.org/) available. This is a powerful Python 2D plotting library. With the built-in support for SQLite in Python it will only take a couple lines of code to visualize your data. 
+
+```python
+import sqlite3
+from matplotlib import dates
+import matplotlib.pyplot as plt
+
+import homeassistant.util.dt as dt
+
+values = []
+timestamps = []
+
+conn = sqlite3.connect('/home/ha/.homeassistant/home-assistant_v2.db')
+data = conn.execute("SELECT state, last_changed FROM states WHERE "
+                    "entity_id = 'sensor.aare' AND last_changed BETWEEN "
+                    "'2016-07-05 00:00:00.000000' AND "
+                    "'2016-07-07 00:00:00.000000'")
+
+for x in data:
+    timestamps.append(dates.date2num(dt.parse_datetime(x[1])))
+    values.append(float(x[0]))
+
+plt.plot_date(x=timestamps, y=values, fmt="r-")
+plt.ylabel('Temperature')
+plt.xlabel('Time line')
+
+plt.savefig('sensor.png')
+```
+
+Creating a connection to the database and executing a query is similar to the ways already seen. The return values from the query are splitted into two lists. The time stamps must be converted in an value which is accepted by matplotlib and then the graph is generated and saved as image. 
+
+<p class='img'>
+  <img src='/images/blog/2016-07-reporting/mpl-sensor.png' />
+  Sensor graph generated by matplotlib
+</p>
+
+Most of the graphs are pretty ugly. So, further beautification will be needed. If you have created a nice report including some amazing graphs then the Home Assistant community would be grateful for sharing them in our [forum](https://community.home-assistant.io/). 
+
+Perhaps will a follow-up of this blog post cover [Jupyter Notebook](http://jupyter.org/). Those notebooks are an easy and interactive way to process stored information.
```

Comment:
I think we can skip this line and just publish something about that in the future 👍